File: test/language/testdata/typescript/StepDefinitions.ts
import { Given } from '@cucumber/cucumber'
import assert from 'assert'

Given('a {uuid}', async function (uuid: string) {
  assert(uuid)
})

Given('a {date}', async function (date: Date) {
  assert(date)
})

Given('a {planet}', async function (planet: string) {
  assert(planet)
})

Given(/^a regexp$/, async function () {
  assert(true)
})

Given('an {undefined-parameter}', async function (date: Date) {
  assert(date)
})

Given("the bee's knees", async function () {
  assert(true)
})

File: test/language/testdata/scala/StepDefinitions.scala
import io.cucumber.scala.{EN, ScalaDsl, Scenario}
import org.junit.Assert._

class StepDefinitions extends ScalaDsl with EN {

  Given("a {uuid}") { (uuid: String) =>
  }

  When("a {date}") { (date: Date) =>
  }

  Then("a {planet}") { (date: Date) =>
  }

  Then("""^a regexp$""") { (x: Date) =>
  }

  But("an {undefined-parameter}") { (date: Date) =>
  }

  Given("the bee's knees") { (date: Date) =>
  }
}
